libs/asset-editor/src/lib/components/asset-editor-tabs/asset-editor-files/asset-editor-files.component.ts

```diff
@@ -1,12 +1,249 @@
+    const input = event.target as HTMLInputElement;
+    if (input.files && input.files.length > 0) {
+      this.handleFiles(input.files);
+    }
+  }
+
+  private handleFiles(files: FileList) {
+    for (const element of Array.from(files)) {
+      if (element.size > 2000 * 1024 * 1024) {
+        this.isFileTooLarge = true;
+        return;
+      }
+      this.isFileTooLarge = false;
+      this.form.controls.assetFiles.push(
+        new FormControl(
+          {
+            id: 0,
+            name: element.name,
+            size: element.size,
+            legalDocItemCode: null as unknown as LegalDocItemCode,
+            type: this.fileType$.value,
+            selected: false,
+            willBeDeleted: false,
+            file: element,
+            lastModifiedAt: new Date(),
+          },
+          { nonNullable: true },
+        ),
+      );
+      this.form.markAsDirty();
+    }
+  }
+
+  public setSearchTerm(term: string) {
+    this.searchTerm$.next(term);
+  }
+
+  public switchFileType(type: AssetFileType) {
+    this.fileType$.next(type);
+  }
+
+  public updateLegalDocItemCode(control: FormControl<FormAssetFile>, event: LegalDocItemCode[]) {
+    control.value.legalDocItemCode = event[0];
+    this.form.markAsDirty();
+  }
+
+  public toggleAll(event: MatCheckboxChange) {
+    this.form.controls.assetFiles.value.forEach((file) => {
+      file.selected = file.type === this.fileType$.value ? event.checked : file.selected;
+    });
+  }
+
+  public selectFile(control: FormControl<FormAssetFile>, event: MatCheckboxChange) {
+    control.value.selected = event.checked;
+  }
+
+  public selectFilesForDeletion() {
+    this.form.controls.assetFiles.setValue(
+      this.form.controls.assetFiles.value.map((file) => {
+        if (file.type !== this.fileType$.value) {
+          return file;
+        }
+        return {
+          ...file,
+          willBeDeleted: file.selected ? true : file.willBeDeleted,
+        };
+      }),
+    );
+    this.form.markAsDirty();
+  }
+
+  public downloadSelectedFiles() {
+    if (!this.assetId) {
+      return;
+    }
+    const filesToDownload: SimpleFile[] = this.dataSource.data
+      .filter((file) => file.value.selected && file.value.id !== 0)
+      .map((file) => ({
+        name: file.value.name,
+        id: file.value.id,
+      }));
+
+    const downloadRequests: Observable<FileBlob>[] = filesToDownload.map((file) => {
+      this.activeFileDownload.add(file.id);
+      return this.httpClient.get(`/api/assets/${this.assetId}/files/${file.id}`, { responseType: 'blob' }).pipe(
+        map((blob) => {
+          return {
+            blob,
+            file,
+          };
+        }),
+      );
+    });
+
+    forkJoin(downloadRequests).subscribe({
+      next: (fileBlobs: FileBlob[]) => {
+        fileBlobs.forEach(async (fileBlob: FileBlob) => {
+          await triggerDownload(fileBlob);
+          this.activeFileDownload.delete(fileBlob.file.id);
+        });
+      },
+    });
```

Comment:
Just use the function directly.
```suggestion
    forkJoin(downloadRequests).subscribe((fileBlobs: FileBlob[]) => {
      fileBlobs.forEach(async (fileBlob: FileBlob) => {
        await triggerDownload(fileBlob);
        this.activeFileDownload.delete(fileBlob.file.id);
      });
    });
```